docker/docker_image_src.go

```diff
@@ -147,7 +147,7 @@ func (s *dockerImageSource) getExternalBlob(ctx context.Context, urls []string)
 			break
 		}
 	}
-	if resp.Body != nil && err == nil {
+	if resp != nil && resp.Body != nil && err == nil {
```

Comment:
Thanks; would a more conventional
```go
if err != nil {
	return nil, 0, err
}
return resp.Body, getBlobSize(resp), nil
```
work just as well? AFAICS `http.Client.Do` always returns a valid `resp.Body` if `err == nil`, so that check was always redundant if any connection was tried — and returning `nil, 0, nil` would not be handled by callers.

Maybe, to be extra careful, add a separate check for
```go
if len(urls) == 0 {
	return errors.New("Internal error: getExternalBlobs called with no URLs")
}
```
at the top, but I don’t think it’s necessary.

Reply:
Yes, that looks better.